app/services/fetch_user_bid_ask_currency_service.rb

```diff
@@ -10,13 +10,39 @@ def call
     @user_currencies_service = user.user_currencies
   
     @user_currencies_service.each do |ucs|
-      buy_sell_currency = BidAskCurrency.find_by(currency_id: ucs['currency_id'])
+      if (buy_sell_currency = BidAskCurrency.find_by(currency_id: ucs['currency_id']) and ucs['purchase'].present?)
```

Comment:
Przy find to rzeczywiście zadziała, ale ogólnie nie warto używać przypisania w warunkach.
W momencie użycia where zamiast find to może zwrócić  pusty array i wtedy też byłoby true
Warto to zostawić u góry tak jak było i do ifa dać tylko:
```suggestion
      if (buy_sell_currency && ucs['purchase'].present?)
```
Dobrą praktyką jest też używanie && i || zamiast and i or